### PR TITLE
Several significant changes

### PR DESCRIPTION
**Pull request by elight**
- More automation; reduces the amount of work to define a Domain Model class
- Removed ORM abstraction layer; edr is now dedicated specifically to AR
- Minor refactorings

More API changes and refactorings to come.

I'm uncertain whether this deviates enough from the original version to warrant a full blown fork/separate gem.  I'd say it depends upon how far it deviates from your goals for the gem.

### CODE REVIEW

File: lib/edr/ar/repository.rb [hunk 1]
```diff
@@ -1,17 +0,0 @@
-module Edr
-  module AR
-    module Repository
-      include ::Edr::Repository
```
Comment (vsavkin) [inline lib/edr/ar/repository.rb:4]: We use EDR with different ORMs (such as Sequel), so removing the ORM layer doesn't really work for us. Why is it necessary?

File: lib/edr/registry.rb [hunk 2]
```diff
@@ -3,21 +3,40 @@
...
-      @@instance.data_class_for(model_class)
+      @instance.data_class_for(model_class)
     end
 
     def self.model_class_for(data_class)
-      @@instance.model_class_for(data_class)
+      @instance.model_class_for(data_class)
     end
 
+    def self.map_models_to_mappers
```
Comment (vsavkin) [inline lib/edr/registry.rb:17]: Not sure if using naming conventions will work in all the cases. Consider the case when the same data object is used by multiple models, or when some legacy class is used as a data class.

File: spec/test_data.rb [hunk 3]
```diff
@@ -46,37 +46,25 @@ class ItemData < ActiveRecord::Base
...
-
   def add_item attrs
     repository.create_item self, attrs
   end
 end
 
 class Item
-  include Edr::Model
-
-  fields :id, :name, :amount, :order_id
```
Comment (vsavkin) [inline spec/test_data.rb:62]: I think a model should declare its fields. Pulling them automagically from the corresponding data object is very implicit.

File: lib/edr/registry.rb (hunk 2 above)
Comment (elight) [inline lib/edr/registry.rb:17]: That's the thing: that's not the 80% case. The average developer won't encounter that situation.

File: spec/test_data.rb (hunk 3 above)
Comment (elight) [inline spec/test_data.rb:62]: Yes it is. It's also the Rails Way.

It's ok. I'll fork the project, rename it, credit you and the other precious devs, and carry on in a different direction.

As I wrote in the PR, I suspected that my approach deviates from where you were going. But my intended audience, I believe, is also different.

Comment (vsavkin) [inline spec/test_data.rb:62]: Yup, I think our approaches are rather different and forking the project is the way to go.

Comment (elight) [inline spec/test_data.rb:62]: Rock on. Pleasure having a rational OSS discussion with someone. :thumbsup:

**Pull request closed without merging**